library/Zend/Log/Writer/Mail.php

```diff
@@ -86,6 +86,9 @@ public function __construct($mail, Transport\TransportInterface $transport = nul
             }
             $transport = isset($mail['transport']) ? $mail['transport'] : null;
             $mail      = isset($mail['mail']) ? $mail['mail'] : null;
+            if (is_array($mail)) {
```

Comment:
You don't need this verification, because you're already inside of a `is_array($mail)`